scripts/testing/e2e-runner

```diff
@@ -0,0 +1,348 @@
+    fi
+
+    info "removing worktree $WORKTREE..."
+    cd $CLONED_REPO
+    if ! git worktree remove $WORKTREE; then
+        fatal "failed to remove worktree $WORKTREE..."
+    fi
+}
+
+generate_summary() {
+    local summary="$RESULT_DIR/summary.txt"
+    local dir test result failed
+
+    info "generating summary $summary..."
+    rm -f $summary
+    if ! touch "$summary"; then
+        error "failed to create $summary"
+        return
+    fi
+
+    for dir in $(find $RESULT_DIR/$TEST_SUITE_DIR -name 'test[0-9]*-*' -type d); do
+        test="${dir##*$TEST_SUITE_DIR/}"
+        result="FAIL"
+        if [ -f "$dir/summary.txt" ]; then
+            if grep -q PASS "$dir/summary.txt"; then
+            result="PASS"
+            fi
+        fi
+        if [ "$result" = "FAIL" ]; then
+            failed=true
+        fi
+        echo "$result $test"
+    done | sort > $summary
+}
+
+generate_index() {
+    local index="$RESULT_ROOT/index.html"
+    local run status
+
+    info "updating index $index..."
+    rm -f $index
+    if ! touch "$index"; then
+        error "failed to create $index"
+        return 1
+    fi
+
+    >>$index cat <<EOF
+    <html>
+      <head><title>NRI Reference Plugins Test Results</title></head>
+      <body>
+EOF
+
+    echo "      <ul>" >> $index
+    for run in $(ls -1 $RESULT_ROOT --sort=time); do
+        case $run in
+            [0-9][0-9][0-9][0-9]-*) ;;
+             *) continue
+                ;;
+        esac
+        if [ ! -e $RESULT_ROOT/$run/summary.txt ] || grep -q FAIL $RESULT_ROOT/$run/summary.txt; then
+            status="FAIL"
+        else
+            status="PASS"
+        fi
+        echo "        <li>$status <a href=\"$run\">$run</a></li>" >> $index
+    done
+    echo "      </ul>" >> $index
+
+    >>$index cat <<EOF
+      </body>
+    </html>
+EOF
+}
+
+#
+# main script
+#
+
+HANDOVER_ARGS=""
+
+while [  "$#" -gt 0 ] && [ "${1#-}" != "$1" ]; do
+    case $1 in
+        --source-proxies)
+            PROXIES_SRC="$2"
+            shift 2
+            ;;
+        --remote|-r)
+            REMOTE_REPO="$2"
+            shift 2
+            ;;
+        --branch|-b)
+            TEST_BRANCH="$2"
+            shift 2
+            ;;
+        --local|-l)
+            CLONED_REPO="$2"
+            shift 2
+            ;;
+        --results|--out|-o)
+            RESULT_ROOT="$2"
+            shift 2
+            ;;
+        --name|-n)
+            RESULT_NAME="$2"
+            shift 2
+            ;;
+        --skip-worktree)
+            SKIP_WORKTREE=1
+            shift
+            ;;
+
+        --trace|-t|-x)
+            set -x
+            shift
+            ;;
+        --help|-h)
+            usage $0
+            ;;
+
+        *)
+            fatal "unknown command line option $1"
+            ;;
+        esac
+done
+
+E2E_TESTS="$*"
+RESULT_DIR="$RESULT_ROOT/$RESULT_NAME"
+
+source_proxies
+prepare_worktree
+prepare_result_dir
+run_e2e_tests
```

Comment:
should we try to cleanup vms also before run_e2e_tests to make sure we have a clean start every time?